### ISSUE TITLE
launcher exits the VM with status code 0 when an exception is thrown VertxApplicationHooks

### ISSUE THREAD

**Issue by bitkid** [eclipse-vertx/vertx-launcher#15]
if i put a throw Exception("bla") into one of the application hooks (f.ex. in afterVertxStarted) the JVM exits with status 0. i think it should not.

**Comment by tsegismont** [comment]
Thanks for reporting this @bitkid

**Comment by tsegismont** [comment]
Please share a small snippet that fails your expectations with 5.0.0.CR3

**Comment by bitkid** [comment]
https://gist.github.com/bitkid/30456f73833b9074e3ac39704256d4e2

this is the simple test code.

i put it into a fat jar and start it with

java -cp the-all.jar Launchie

it writes

"throw exception" in the shell

and when i do echo $? it returns 0

**Comment by bitkid** [comment]
ok, so i looked at your tests .. do i have to do

System.exit(app.launch()) ?

**Comment by bitkid** [comment]
ah ok .. System.exit() is only called when the exit code is 2 .. what's the reason behind that? because if the beforeStartingVertx throws  exit code is 1

**Comment by tsegismont** [comment]
Good catch. Indeed, we have a bug in the latest version, `System.exit` is only invoked for usage (code 2) errors.
